setup.py

```diff
@@ -30,13 +26,16 @@
     description=description,
     long_description=description,
     url='https://github.com/cloudera/thrift_sasl',
+    setup_requires=['setuptools>=36.2'],
```

Comment:
Is this requirement necessary? Note that with python3.5 virtual environments come with setuptools==28.8.0 by default. Of course users can always upgrade to the latest version of setuptools but things would be easier if they didn't have to upgrade first. What do you think?

Reply:
Hmm - I think you are right. `v36.2` is much higher than necessary I think, but it is quite hard to tell from the documentation and package histories that are available on the web.

Digging into this a bit more:

- per the [packaging documentation history](https://packaging.pypa.io/_/downloads/en/latest/pdf/), it looks like PEP 508 environment marker support was added in `packaging@16.0`.
- per the [setuptools history](https://setuptools.readthedocs.io/en/latest/history.html#id697) and in digging through the [setuptools source code](https://github.com/pypa/setuptools/blob/20.2/pkg_resources/_vendor/packaging/__about__.py), the earliest evidence of them using `packaging>=16.0` that I can see was in `v20.2` but it seems that environment markers in this version are only supported via `extras_require` (?)
- per the [setuptools history](https://setuptools.readthedocs.io/en/latest/history.html#id689), I believe that support for environment markers in `install_requires` was added in `v20.5`

So from what I can tell, it seems that as long as a user has a version of `setuptools>=20.5` the environment markers should be handled correctly. 

If you are comfortable with it, we can either:

- drop the `setup_requires` line, since it would seem that most projects are likely using a high enough `setuptools` to handle environment markers in `install_requires`
- adjust to `setup_requires=['setuptools>=20.5']`

Let me know which you prefer and I will make the adjustment!

Reply:
Thanks for the analysis. I'd prefer adding "setup_requires=['setuptools>=20.5']".